### PR TITLE
docs: add docs for functions

### PR DESCRIPTION
**Pull request by Yisaer**

### CODE REVIEW

File: docs/en_US/sqls/built-in_functions.md [hunk 1]
```diff
@@ -179,6 +179,21 @@ When casting to datetime type, the supported column type and casting rule are:
 
 Currently, 'zlib', 'gzip', 'flate' and 'zstd' method are supported.
 
+## Array Functions
+
+| Function   | Example                   | Description                                                           |
+|------------|---------------------------|-----------------------------------------------------------------------|
+| array_position | array_postion(array, value) | Return a 1-based index of the first occurrence of val if it is found within array. If val does not exist within array, it returns 0 |
```
Comment (ngjaying) [inline docs/en_US/sqls/built-in_functions.md:186]: 1-based, is it better to be 0-based?

Comment (Yisaer) [reply docs/en_US/sqls/built-in_functions.md:186]: Both are ok to me, for now rockset used 1-based.

Comment (ngjaying) [reply docs/en_US/sqls/built-in_functions.md:186]: I prefer 0-based because currently all our indexes are 0-based. It is weird to have 1 based here.

Comment (Yisaer) [reply docs/en_US/sqls/built-in_functions.md]: updated

File: docs/en_US/sqls/built-in_functions.md [hunk 2]
```diff
@@ -179,6 +179,21 @@ When casting to datetime type, the supported column type and casting rule are:
 
 Currently, 'zlib', 'gzip', 'flate' and 'zstd' method are supported.
 
+## Array Functions
+
+| Function   | Example                   | Description                                                           |
+|------------|---------------------------|-----------------------------------------------------------------------|
+| array_position | array_postion(array, value) | Return a 0-based index of the first occurrence of val if it is found within array. If val does not exist within array, it returns 0 |
```
[the remark this answers is not in the document]
Comment (Yisaer) [reply docs/en_US/sqls/built-in_functions.md]: updated.

**Review by ngjaying — APPROVED** [review]

**Pull request merged**